VideoWeb/VideoWeb.EventHub/Hub/EventHubClient.cs

```diff
@@ -383,5 +403,59 @@ await Clients.Group(judge.Username.ToLowerInvariant())
                 _logger.LogError(ex, "Error occured when updating participant device status");
             }
         }
+
+        private async Task<string> GetParticipantUsernameByIdAsync(Guid conferenceId, string participantId)
+        {
+          var username = string.Empty;
+            try
+            {
+                var participantGuidId = Guid.Parse(participantId);
+                var conference = await GetConference(conferenceId);
+                var participant = conference.Participants.SingleOrDefault(x =>
+                    x.Id == participantGuidId);
+
+                if (participant == null)
+                {
+                    _logger.LogDebug($"Participant {participantId} does not exist in conversation");
+                    throw new ParticipantNotFoundException(conferenceId, Context.User.Identity.Name);
+                }
+
+                username = participant.Username;
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, "Error occured when validating send message");
```

Comment:
was this also to catch the "SignleOrDefault" finding multiple? or just for the ParticipantNotFoundException? if just ParticipantNotFoundException maybe we could remove the catch and just return there instead of throwing?

Reply:
Refactoring code